train.py

```
@@ -303,11 +316,14 @@ def main() -> int:
-        from sidestep_engine._compat import check_compatibility
-        check_compatibility()
-    except Exception:
-        pass  # never let the compat check itself crash the CLI
+    # Skip for lightweight subcommands to avoid importing torch/PEFT/Lightning
+    # (~2-3 GB RAM) when all we need is json + pathlib.
+    if not _is_lightweight_subcommand():
+        try:
+            from sidestep_engine._compat import check_compatibility
+            check_compatibility()
+        except Exception:
+            pass  # never let the compat check itself crash the CLI
```

Ruff (0.16.0)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)